src/ReinforcementLearningCore/src/policies/agent/multi_agent.jl

```diff
@@ -188,13 +188,24 @@ function Base.push!(multiagent::MultiAgentPolicy, ::PreActStage, env::E) where {
     end
 end
 
+# Multiagent Version
+function RLBase.plan!(agent::Agent{P,T,C}, env::E, p::Symbol) where {P,T,C,E<:AbstractEnv}
+    action = RLBase.plan!(agent.policy, env, p)
+    push!(agent.cache, action)
+    action
+end
+
 # Like in the single-agent case, push! at the PostActStage() calls push! on each player with the reward and termination status of the environment
 function Base.push!(multiagent::MultiAgentPolicy, ::PostActStage, env::E) where {E<:AbstractEnv}
```

Comment:
multiagent[i].trajectory only exists, if the policies are of type Agent, right? In my opinion, the focus should be first on single agent and afterward thinking about multi-agent case.